### PR TITLE
XWIKI-18363: Auto-detection of language with the code macro is broken

### PR DESCRIPTION
**Pull request by manuelleduc**
https://jira.xwiki.org/browse/XWIKI-18363

- Downgrade Pygments from 2.5.1 to 2.4.2 because of regressions in the way Pygment guesses the language of the code.
- Introduction of two non regression tests for future upgrades.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/pom.xml [hunk 1]
```diff
@@ -52,7 +52,7 @@
     <dependency>
       <groupId>org.pygments</groupId>
       <artifactId>pygments</artifactId>
-      <version>2.5.1</version>
+      <version>2.4.2</version>
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/pom.xml:55]: I'd put a comment and a link to the explanation to why we should upgrade to 2.5+ (2 reasons)

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode15.test [hunk 2]
```diff
@@ -0,0 +1,184 @@
+.runTransformations
+.#-----------------------------------------------------
+.input|xwiki/2.0
+.# Validates that the code macro accurately identifies the language used in the code macro when not specified.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode15.test:4]: I'd explain more why this test.

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode15.test [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+.#-----------------------------------------------------
+{{code}}
+    <!-- disable to use ajp connector instead
+    <Connector port="8080" protocol="HTTP/1.1"
+               connectionTimeout="20000"
+               URIEncoding="UTF-8"
+               redirectPort="8443" />
+    -->
+{{/code}}
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode15.test:13]: you don't need the extra leading spaces

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test [hunk 4]
```diff
@@ -0,0 +1,147 @@
+.runTransformations
+.#-----------------------------------------------------
+.input|xwiki/2.0
+.# Validates that the code macro accurately identifies the language used in the code macro when not specified.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test:4]: Same this needs to explain why it's different from the other 2 tests with the same comment

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/pom.xml [hunk 5]
```diff
@@ -49,6 +49,11 @@
       <groupId>org.python</groupId>
       <artifactId>jython-slim</artifactId>
     </dependency>
+
+    <!-- Pygments must stay in version < 2.5.x because this version and above have issues with the auto-detection of 
+         the language of the code (MIME and SQL are too often detected). Pygments 2.8.0+ present a fix, but is only 
+         compatible with Python 3.0, which we do not support for now.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/pom.xml:55]: This is not correct. v2.6.0+ requires python3, see the changeog

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test [hunk 6]
```diff
@@ -2,6 +2,9 @@
 .#-----------------------------------------------------
 .input|xwiki/2.0
 .# Validates that the code macro accurately identifies the language used in the code macro when not specified.
+.# Pygments 2.4.2, and 2.0.8 identify the sample below as Tera Term macro, presenting an acceptable highlight of
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test:5]: >  Tera Term macro

macro? You mean content?

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test:5]: That's the human readable name of the TeraTermLexer.

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test:5]: sure but your sentence doesn't make sense

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/test/resources/macrocode16.test:5]: what you mean is "as Tera Term Macro content". You forgot "content". Same problem with "MIME" you cannot says it's recognized as MIME but as MIME content or syntax.

**Pull request merged**